refactor: Modernize overlay architecture and improve bottom sheet UX

lourou opened the pull request:
This PR modernizes our overlay architecture by partially removing the Portal dependency and improving the bottom sheet implementation. It also fixes the reaction context menu that has not worked since the Expo 52 update because of the use of Portal and incompatibility with our ThemeProvider.

Key Changes:
- Remove the unmaintained `@gorhom/portal` dependency in favor of React Native's Modal (still needed by some other parts of the app)
- Replace `BottomSheetModal` with standard `BottomSheet` wrapped in `Modal`
- Improve bottom sheet behavior and responsiveness:
  - Fix snap points by disabling dynamic sizing
  - Improve dismiss/reopen flow with proper animation handling
  - Fix reactions filter initialization on drawer open

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
  - Enhanced bottom sheet and modal interactions across the application
  - Improved theme-based styling for components
  - Introduced a new `BottomSheet` component for customizable styles

- **Improvements**
  - Updated message context menu display to a modal approach
  - Refined drawer rendering with improved visibility and interaction handling
  - Streamlined component visibility management for message reactions

- **Technical Updates**
  - Removed portal-based rendering in favor of modal components
  - Introduced new state management for message reactions

- **Dependency Changes**
  - Updated `@gorhom/bottom-sheet` package
  - Removed `@gorhom/portal` package
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

thierryskoda commented on features/conversation/conversation-message/conversation-message-reactions/conversation-message-reaction-drawer/conversation-message-reaction-drawer.tsx:
```diff
@@ -1,142 +1,193 @@
-      topInset={insets.top}
-      snapPoints={["50%", "100%"]}
+    <Modal
+      visible={isVisible}
```
Doesn'T make any sense because you have

 if (!isVisible) {
     return null;
   }

Above

lourou commented on design-system/BottomSheet/BottomSheet.tsx:
```diff
@@ -0,0 +1,47 @@
+        <BottomSheet
+          ref={ref}
+          backdropComponent={backdropComponent}
+          handleIndicatorStyle={{
+            backgroundColor: theme.colors.background.raised,
+          }}
+          handleStyle={{
+            backgroundColor: theme.colors.background.raised,
+            borderTopLeftRadius: theme.borderRadius.sm,
+            borderTopRightRadius: theme.borderRadius.sm,
+          }}
+          backgroundStyle={{
+            backgroundColor: theme.colors.background.raised,
+          }}
```
yesè

lourou commented on design-system/BottomSheet/BottomSheet.tsx:
```diff
@@ -0,0 +1,47 @@
+          }}
+          {...rest}
+        >
+          <BottomSheetView style={{ flex: 1 }}>{children}</BottomSheetView>
```
Yes, I've just removed it and use the {children} directly

lourou commented on features/conversation/conversation-message/conversation-message-reactions/conversation-message-reaction-drawer/conversation-message-reaction-drawer.tsx:
```diff
@@ -1,142 +1,193 @@
+                  themed($chip),
+                  filterReactions === null && themed($chipActive),
+                ]}
+                underlayColor={theme.colors.border.subtle}
```
Good call, I imported our own `TouchableHighlight` and simplified the props

lourou commented on features/conversation/conversation-message/conversation-message-context-menu/conversation-message-context-menu-backdrop.tsx:
```diff
@@ -10,8 +11,10 @@ export const MessageContextMenuBackdrop = memo(
+
     return (
-      <BlurView isAbsolute>
+      <BlurView isAbsolute tint={theme.isDark ? "dark" : "light"}>
```
It's now in the component itself :)

lourou commented on features/conversation/conversation-message/conversation-message-reactions/conversation-message-reaction-drawer/conversation-message-reaction-drawer.tsx:
```diff
@@ -1,142 +1,193 @@
+          animateOnMount
+          detached={false}
+          enableDynamicSizing={false}
+          onChange={(index) => {
+            if (index === -1) {
+              handleDismiss();
+            }
```
I do this to make sure to dismiss the wrapping modal when the bottom sheet is dismissed

thierryskoda reviewed (approved):
Small loom for more feedback but approved because it might not be relevant and priority https://www.loom.com/share/d941330aa95c4ccb816cb845d0246a3f

lourou commented on features/conversation/conversation-message/conversation-message-context-menu/conversation-message-context-menu.tsx:
```diff
@@ -124,11 +123,13 @@ const Content = memo(function Content(props: {
+        animationType="fade"
+        statusBarTranslucent={Platform.OS === "android"}
+      >
+        <Animated.View style={StyleSheet.absoluteFill}>
```
Yes 💯

lourou commented:
Thanks! I've simplified the reactions drawer implementation:

1. Removed redundant visibility state:
   - Removed `isVisible` from store
   - Removed local `useState` for visibility
   - Now using `!!rolledUpReactions.totalCount`

2. Improved Modal + BottomSheet integration:
   - Removed unnecessary `onRequestClose` handler
   - Simplified dismiss logic
   - Better animation coordination between Modal and BottomSheet

Code is now more straightforward with fewer moving parts and clearer state management. The drawer shows when reactions are present and dismisses cleanly when the bottom sheet is closed.

thierryskoda reviewed (approved):
🚀
